### ISSUE TITLE
Unexpected @srsName value on element srsName

### ISSUE THREAD

**Issue by CeciliaPH** [opengeospatial/ets-wfs20#277]
**Describe the bug**
When validating our service
https://www.ign.es/wfs-inspire/ngbe?request=GetCapabilities&service=WFS

We receive this error.
Simple WFS (1 error). OGC TEAM Engine reported a failed test: Unexpected @srsName value on element srsName expected [http://www.opengis.net/def/crs/EPSG/0/4326] but found [urn:ogc:def:crs:EPSG:6.3:4326]

We don't understand where this error comes from. If you make the individual requests there is no way that the servers returns  [urn:ogc:def:crs:EPSG:6.3:4326]. We would like to understand where this urn style string is coming from.

Thanks,

Cecilia

**To Reproduce**
https://www.ign.es/wfs-inspire/ngbe?REQUEST=GetFeature&SERVICE=WFS&VERSION=2.0.0&COUNT=10&TYPENAMES=ns41:NamedPlace&SRSName=http://www.opengis.net/def/crs/EPSG/0/4326

https://www.ign.es/wfs-inspire/ngbe?REQUEST=GetFeature&SERVICE=WFS&VERSION=2.0.0&COUNT=10&TYPENAMES=ns41:NamedPlace&SRSName=http://www.opengis.net/def/crs/EPSG/0/25830

**Comment by CeciliaPH** [comment]
We made further investigation and discovered that is something that happens when you retrieve data by a POST request. If you ask via REST request the data retrieved should be correct.
We are using deegree 3 to serve this data. May be is an internal parameter that changes the srsName while using POST. 
Does the petition need to be POST to accomplish the requirement?

**Comment by bpross-52n** [comment]
@CeciliaPH I can confirm your findings, the incorrect EPSG code is returned from the server when requesting using POST. When the service supports POST requests, this will be tested. However, POST is not mandatory for WFS 2.0 services, see [Chapter 6.4 of the WFS 2.0 standard.](https://docs.ogc.org/is/09-025r2/09-025r2.html#24)

**Comment by CeciliaPH** [comment]
Hi Benjamin, 

We are really interested in complete the tests successfully. 
If the test is successful in GET, we understand that validator should not raise an error.
We wonder which should be the next steps, as this issue is in Waiting in CITE.

Best regards,

**Comment by dstenger** [comment]
Hi @CeciliaPH,
The unexpected srsName is produced by your service when requested via POST. Thus, the error seems to be on the service side. Can you please check that?
The request sent by TEAM Engine should be correct. @bpross-52n Can you please share the exact request which is leading to the response with the unexpected srsName?

**Comment by CeciliaPH** [comment]
Exactly, the srsName is produced by deegree 3. There is something intern that changes de srsName style, to the on with urn. Any way, using GET it works properly, but engine uses POST.

This is the kind of post request that raises the error. (any petition would, as the software changes de srsName style)

<wfs:GetFeature xmlns:wfs='http://www.opengis.net/wfs/2.0' count='10' service='WFS' version='2.0.0'>
   <wfs:Query xmlns:ns41='http://inspire.ec.europa.eu/schemas/gn/4.0'
              srsName='http://www.opengis.net/def/crs/EPSG/0/25830'
              typeNames='ns41:NamedPlace'/>
</wfs:GetFeature>

**Comment by dstenger** [comment]
@CeciliaPH Did you solve your problem with deegree 3 or is it still an issue for you? If it is solved, I propose to close this issue.

**Comment by CeciliaPH** [comment]
Hi,

Unfortunately the issue is not solved yet. As the problem comes when yo ask this via POST and not via URL.

**Comment by dstenger** [comment]
I just checked your service with https://cite.ogc.org/teamengine/.

All tests of `Simple WFS` are passing. Thus, you are passing all mandatory conformance classes.

Conformance class `Basic WFS` is failing, including the here reported problem. Thus, you are not passing this optional conformance class.

I compared your result with the one of the [deegree reference implementation](https://cite.deegree.org/deegree-webservices-3.5.6/services/wfs200?service=WFS&amp;request=GetCapabilities).
Here, the test `get Feature In Other CRS` is skipped.
Thus, this might be a problem within deegree.

If you cannot solve the problem, you could contact the deegree community and ask for help there.

**Comment by ghobona** [comment]
Since this was determined to be an issue with deegree 3, this GitHub ticket can be closed.
